freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/stop/StackStopService.java

```diff
@@ -39,6 +43,7 @@ public void finishStackStop(StackStopContext context, StopInstancesResult stopIn
         stackStartStopService.validateResourceResults(context.getCloudContext(),
                 stopInstancesResult.getErrorDetails(), stopInstancesResult.getResults(), false);
         stackUpdater.updateStackStatus(stack.getId(), DetailedStackStatus.STOPPED, "Stack infrastructure stopped successfully.");
+        providerChecker.updateAndGetStatuses(stack, true);
```

Comment:
the `finishStackStop` method is called from an action and not a handler
`ProviderChecker#updateAndGetStatuses` is calling the cloud provider which means it can take significant time (more likely if any retry can happen)
Actions should have only short running logic, like updating the DB.
I would prefer having this called from a handler. (if you think we need this)

Is there any reason why we don't just rewrite the instance status instead of going to the provider? The previous step (I haven't checked the actual code) should wait for the instances to be stopped on the provider, I don't think checking it again has any additional benefit.

Reply:
@lacikaaa   I made the change according to your suggestion. Now instead of polling from the provider, we update the instance status directly. Internally, it waits till the instances stopped and stack status stopped, and then it updates the instance status to stopped at last. Tested again and everything works. Thx!